diff --git a/node_modules/react-native-reanimated/lib/reanimated2/Hooks.js b/node_modules/react-native-reanimated/lib/reanimated2/Hooks.js
index aca4b44..60ef39c 100644
--- a/node_modules/react-native-reanimated/lib/reanimated2/Hooks.js
+++ b/node_modules/react-native-reanimated/lib/reanimated2/Hooks.js
@@ -129,7 +129,8 @@ function runAnimations(animation, timestamp, key, result, animationsActive) {
 function isAnimated(prop) {
     'worklet';
     if (Array.isArray(prop)) {
-        for (const item of prop) {
+      for (let i = 0; i < prop.length; i++) {
+        const item = prop[i];
             for (const key in item) {
                 if (item[key].onFrame !== undefined) {
                     return true;
@@ -454,7 +455,8 @@ export function useAnimatedStyle(updater, dependencies, adapters) {
         let result = false;
         // a case for transform that has a format of an array of objects
         if (Array.isArray(element)) {
-            for (const elementArrayItem of element) {
+            for (let i = 0; element.length < i; i++) {
+              const elementArrayItem = element[i];
                 // this means unhandled format and it doesn't match the transform format
                 if (typeof elementArrayItem !== 'object') {
                     break;
